packages/@dcos/sorting/index.ts

```diff
@@ -0,0 +1,61 @@
+export enum ComparisonResult {
+  BIGGER = 1,
+  EQUAL = 0,
+  SMALLER = -1
+}
+export type Comparator<T> = (a: T, b: T) => ComparisonResult;
+export type Lense<T> = (data: T) => any;
+
+export function sort<T>(comps: Array<Comparator<T>>, data: T[]): T[] {
+  return [...data].sort((a, b) => {
+    for (const comperator of comps) {
+      const result = comperator(a, b);
+
+      if (result !== 0) {
+        return result;
+      }
+    }
+
+    return 0;
+  });
+}
+
+export function reverseComparator<T>(comparator: Comparator<T>): Comparator<T> {
+  return (a, b) => {
+    const previousResult = comparator(a, b);
+
+    if (previousResult === ComparisonResult.EQUAL) {
+      return previousResult;
+    } else {
+      return (-1 * previousResult) as ComparisonResult;
+    }
+  };
+}
+
+function compareStringValues(a: any, b: any): ComparisonResult {
+  if (a == null || b == null || a === b) {
+    return ComparisonResult.EQUAL;
+  }
+
+  return a < b ? ComparisonResult.SMALLER : ComparisonResult.BIGGER;
+}
+
+function compareNumericValues(a: any, b: any): ComparisonResult {
+  const aNumber = parseInt(a, 10);
+  const bNumber = parseInt(b, 10);
+
+  if (isNaN(aNumber) || isNaN(bNumber) || aNumber === bNumber) {
```

Comment:
I think we should have tests mixing NaN, null, and undefined in the list.

Reply:
With tests you mean unit tests and not checks, right?

Reply:
👍